core/primitives/src/transaction.rs

```diff
@@ -72,6 +72,14 @@ impl Action {
             _ => 0,
         }
     }
+    #[cfg(feature = "protocol_feature_tx_size_limit")]
```

Comment:
I suggest for the transaction size to use the size of the transaction serialized as borsh. This will remove potential tricky angles of attack, e.g. it is still possible to create a heavy transaction that adds a very large number of function access keys with long method names. Also, that removes ambiguity in interpreting what transaction size is, so e.g. we can say that block size is sum of the transaction sizes plus header data.

Ideally, we would add a trait to Borsh called `BorshSize` that implements `borsh_size() -> u64` method which we can use similarly to `BorshSerialize` and `BorshDeserialize`, like this: `#[derive(BorshSize)]`. AFAIR there might some places in our code where we measure the size of the objects by first serializing them in Borsh and then counting bytes, so it might be useful there too. It shouldn't be difficult to add to borsh and will probably be largely copy-paste of https://github.com/near/borsh-rs/blob/master/borsh/src/ser/mod.rs

Reply:
Yeah agree with what Max said here

Reply:
Thank you!
Changed the way of size computation and added a separate issue for adding new trait.